### DISCUSSION TITLE
Why does batch_size in dataloader only create batches across graphs, and not across data samples?

### DISCUSSION THREAD

**Discussion by sidhantls** [pyg-team/pytorch_geometric#2634] [Q&A] [answered]
Is there any reason why the dataloader creates batches only across graphs but not across the data samples? 

I'm using the WikiCS dataset, and using a dataloader here. This dataset only contains one graph.  Because it has only 1 graph, when I'm using the dataloader, it only creates on batch of the same size of the dataset. If a dataset just contains one large graph, then wouldn't it be desired to create batches from this? 

```
dataset = torch_geometric.datasets.WikiCS(data_dir)
print(dataset) # WikiCS()

loader = DataLoader(dataset, batch_size=16, shuffle=True)
print(len(loader)) # 1 

for batch in loader:
    break
print(batch) # Batch(batch=[11701], edge_index=[2, 297110], ptr=[2], stopping_mask=[11701, 20], test_mask=[11701], train_mask=[11701, 20], val_mask=[11701, 20], x=[11701, 300], y=[11701])
```

**Comment by ChenYizhu97** [comment, marked as answer]
For the dataset consists of one large graph, I think you might want to divide the graph into several partitions first, then use a proper dataloader such as Clusterloader so that each batch contains several partitions of the graph. By doing this, you train the model on a subset of the graph each step.

[This notebook](https://colab.research.google.com/drive/1XAjcjRHrSR_ypCk_feIWFbcBKyT4Lirs?usp=sharing#scrollTo=RwuwXWLZ_4OQ) provided by torch_geometric [document](https://pytorch-geometric.readthedocs.io/en/latest/notes/colabs.html) would be helpful.

**Reply by rusty1s** [reply]
Yes, this is correct. Note that there exists multiple techniques for creating mini-batches of single graphs. As such, `DataLoader` refers to creating mini-batches of a set of graphs, while other dataloaders try to tackle the task of creating mini-batches of a single graph, e.g., `NeighborSampler`, `ClusterLoader`, `GraphSAINTSampler` or ` ShaDowKHopSampler`.
